e2e/integration_tests/data_flow_test.go

```diff
@@ -326,10 +331,9 @@ func Test_RetrieveCapabilityStatements(t *testing.T) {
 	th.Assert(t, mq != nil, "expected message queue to be created")
 	th.Assert(t, chID != nil, "expected channel ID to be created")
 
-	ctx, cancel := context.WithCancel(context.Background())
+	ctx, _ := context.WithTimeout(context.Background(), 30 * time.Second)
 	go capabilityhandler.ReceiveCapabilityStatements(ctx, store, mq, chID, qName)
 	time.Sleep(30 * time.Second)
```

Comment:
i believe that instead of doing the `time.Sleep` here, you'd want to wait for the context to be completed:

```golang
		select {
		case <-ctx.Done():
			// continue after context complete
		}
```

(untested)

I believe this code waits. you'd add the `default` flag if you didn't want it to wait.